client/src/app/shared/models/base.model.ts

```diff
@@ -61,4 +61,58 @@ export abstract class BaseModel extends OpenSlidesComponent implements Deseriali
         Object.assign(this, input);
         return this;
     }
+
+    /**
+     * Get referenced models by ids of the type given by collection.
+     *
+     * Usefull for 1:n/m:n relations, where this model has an array of foreign keys.
+     *
+     * @example
+     * ```ts
+     * groups_id = [1, 2, 5];
+     * groups: Group[] = getReferenceByIds<Group>(Group, group_ids);
+     * --> groups = [<Group1>, <Group2>, <Group5>];
+     * ```
+     *
+     * @param collection the collection of the foreign model.
+     * @param ids the ids of the foreign objects.
+     */
+    protected getReferenceByIds<T extends BaseModel>(collection: ModelConstructor | string, ids: number[]): T[] {
+        if (ids.length === 0) {
+            return [];
+        }
+        const models = this.DS.get(collection, ...ids);
+        if (!models) {
+            return [] as T[];
+        } else if (models instanceof BaseModel) {
+            return [models] as T[];
+        } else {
+            return models as T[];
+        }
+    }
+
+    /**
+     * Get a referenced model by it's id.
+     *
+     * @example
+     * ```ts
+     * category_id = 3;
+     * category: Category = getReferenceById<Category>(Category, category_id);
+     * --> category = <Category3>;
+     * ```
+     *
+     * @param collection The collection of the foreign model.
+     * @param id the id of the foreign object.
+     */
+    protected getReferenceById<T extends BaseModel>(collection: ModelConstructor | string, id: number): T {
```

Comment:
Should go into an own service or be possible over DataStore only.
Base objects should not provide direct access into DataStore over native methods.
Please avoid god patterns

https://en.wikipedia.org/wiki/Single_responsibility_principle

Reply:
I'll move this into the DS.